### PR TITLE
Move tag editor JS into separate file

### PR DESCRIPTION
**Pull request by salty-horse**
I named the file `viewgame.js`, as I think most JS in `viewgame` should move there eventually.

### CODE REVIEW

File: www/viewgame.js [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+
+
+    dispTagTable(tableID, lst, editor, deleteTags)
+    {
+        var tbl = document.getElementById(tableID);
+        tbl.innerHTML = "";
+        var row = null, rownum = 0;
+        for (var i = 0 ; i < lst.length ; i++) {
+            var t = encodeHTML(lst[i].tag);
+            var u = encodeURI8859(lst[i].tag);
```
Comment (dfabulich) [inline www/viewgame.js:66]: Replace `encodeURI8859` (one more spot we won't have to fix later)

File: www/viewgame.js [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+            var ct = lst[i].tagcnt;
+            var cg = lst[i].gamecnt;
+            var m = lst[i].isMine;
+            var n = lst[i].isNew;
```
Comment (dfabulich) [inline www/viewgame.js:70]: All these single-letter variable names make this code hard to read (I'll merge it anyway, though)

**Review by dfabulich — CHANGES_REQUESTED** [review]

File: www/viewgame.js [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+            var cell = document.createElement("div");
+            tbl.appendChild(cell);
+            var s;
+            const ck = "tagCk" + i;
+            if (editor) {
+                const deleteCls = deleteTags ? ' ckdelete' : '';
+                s = `<label class="cklabel" data-ck='${i}'>`
+                    + `<input type="checkbox" class="ckbox${deleteCls}" id="ckBox${ck}"><div class="ckboxImg" aria-hidden="true"></div> `
+                    + `<span>${t}</span></label>&nbsp;<span class=details id="tagCnt${i}">`;
```
[the remark this answers is not in the document]
Comment (salty-horse) [reply www/viewgame.js:79]: Doesn't it add whitepsace between elements that is sometimes significant in HTML?

File: www/viewgame.js [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+function trim(str) {
+    return str.replace(/^\s+|\s+$/g, '');
+}
+
+function tagSorter(a, b)
+{
+    a = a.tag.toLowerCase();
+    b = b.tag.toLowerCase();
+    return (a > b ? 1 : a < b ? -1 : 0);
+}
```
[the remark this answers is not in the document]
Comment (salty-horse) [reply www/viewgame.js:12]: I keep looking for a equivalent of Python's `sort(key=lambda t: t.tag.lower())` but this an improvement :)

File: www/viewgame.js [hunk 5]
```diff
@@ -0,0 +1,317 @@
...
+
+class TagTable {
```
[the remark this answers is not in the document]
Comment (salty-horse) [reply www/viewgame.js:14]: I'm new to modules. How do I pass in the `gameid` and `tagList` into it from the HTML file?

File: www/viewgame.js [hunk 6]
```diff
@@ -0,0 +1,317 @@
...
+            const ck = "tagCk" + i;
+            if (editor) {
+                const deleteCls = deleteTags ? ' ckdelete' : '';
+                s = `<label class="cklabel" data-ck='${i}'>`
+                    + `<input type="checkbox" class="ckbox${deleteCls}" id="ckBox${ck}"><div class="ckboxImg" aria-hidden="true"></div> `
+                    + `<span>${t}</span></label>&nbsp;<span class=details id="tagCnt${i}">`;
+            }
+            else
+                s = "<span class=details title=\"Search for games tagged with "
+                    + t.replace(/"/g, "&#34;")
```
Comment (salty-horse) [inline www/viewgame.js:83]: I don't think this is needed, as `t` is already `encodeHTML()`d

File: www/viewgame.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (dfabulich) [reply www/viewgame.js:14]: Here's one way:
```
<script type=module src=viewgame.js></script>
<script type=module nonce='<?= $nonce ?>'>
import {whatever} from './viewgame.js'
whatever(<?= $gameid ?>)
</script>
```

File: www/viewgame.js [hunk 7]
```diff
@@ -0,0 +1,317 @@
...
+        {
+            if (t.tagcnt != 0)
+                this.dbTagList.push(t);
+            if (t.isMine)
+                tags.push(t.tag);
+        }
+        this.dispTags();
+        this.closeTags("tagEditor");
+        jsonSend("taggame", "tagStatusSpan", this.cbSaveTags.bind(this),
+            {"id": this.gameid, "tags": tags}, true);
```
[the remark this answers is not in the document]
Comment (salty-horse) [reply www/viewgame.js]: That would require changes to jsonSend, no? Maybe best to do separately.

File: www/viewgame.js [hunk 8]
```diff
@@ -0,0 +1,300 @@
...
+
+function dispDeleteTags()
+{
+    document.getElementById("tagStatusSpan").innerHTML = "";
+    dispTagTable("deleteTagTable", memTagList, true, true);
+}
+
+function deleteTag(tag)
+{
+    for (const [i, t] in Object.entries(memTagList))
```
Comment (dfabulich) [inline www/viewgame.js:154]: Deleting tags isn't working for me. I think this `in` should be `of`

**Review by dfabulich — CHANGES_REQUESTED** [review]

File: www/viewgame.js (hunk 8 above)
Comment (dfabulich) [reply www/viewgame.js:154]: Also, in this branch, the "Delete Tags" dialog is missing its "Cancel" button.

**Comment by salty-horse** [comment]
Added fixes in the latest commit, and fixed indentation in html.

**Comment by salty-horse** [comment]
Whoops, looks like this generates `<script1 ` in the html. Will fix.

**Comment by salty-horse** [comment]
Fixed in 625a77344030a6b050f857cb3c33a9c2a5f301ff

**Pull request merged**